### PR TITLE
chore: move display of resource manager limits in daemon startup

### LINKED ISSUE #9802: Kubo writes to stdout when used as a library
### Checklist

- [X] This is a bug report, not a question. Ask questions on [discuss.ipfs.io](https://discuss.ipfs.io).
- [X] I have searched on the [issue tracker](https://github.com/ipfs/kubo/issues?q=is%3Aissue) for my bug.
- [X] I am running the latest [kubo version](https://dist.ipfs.tech/#kubo) or have an issue updating.

### Installation method

built from source

### Version

```Text
github.com/ipfs/kubo v0.18.2-0.20230214211946-0ff406170dd9
```

### Config

_No response_

### Description

When using Kubo as a library, the code at https://github.com/ipfs/kubo/blob/1958510b3e7b796b17cee27fa3424d7310905eae/core/node/libp2p/rcmgr.go#L65-L66 prints to the following to stdout and it isn't really appropriate or useful to our users.

```
Computed default go-libp2p Resource Manager limits based on:
    - 'Swarm.ResourceMgr.MaxMemory': "13 GB"
    - 'Swarm.ResourceMgr.MaxFileDescriptors': 46080

Theses can be inspected with 'ipfs swarm resources'.
```

Happy to send a PR to change it to a log message unless there's a useful way to determine that it is running as a binary and that writing to stdout makes sense then.

### PR DESCRIPTION
**Pull request by gfanton**
Fix #9802 

Move the display of resource manager limits from `core/node/libp2p/rcmgr.go` to `cmd/ipfs/daemon.go` to avoid the use of stdout when kubo is used as library.

### CODE REVIEW

**Comment by BigLep** [comment]
Thanks @gfanton .  The team has been mostly tied up with IFPS Thing 2023 and post-vacation.  I have this down for the 0.21 release.

File: cmd/ipfs/daemon.go [hunk 1]
```diff
@@ -759,6 +764,43 @@ func rewriteMaddrToUseLocalhostIfItsAny(maddr ma.Multiaddr) ma.Multiaddr {
...
+// printRessourceManagerLimit fetches the user-defined resource overrides, and then
+// displays the limits for MaxMemory and MaxFileDescriptors
+func printRessourceManagerLimit(bcfg *node.BuildCfg, cfg *config.Config) error {
+	enabled := cfg.Swarm.ResourceMgr.Enabled.WithDefault(true)
+	switch os.Getenv("LIBP2P_RCMGR") {
+	case "0", "false":
+		enabled = false
+	case "1", "true":
+		enabled = true
+	}
```
Comment (Jorropo) [inline cmd/ipfs/daemon.go:776]: https://github.com/ipfs/kubo/blob/e4908a01633582e205b26ab36769830fe8490c10/core/node/libp2p/rcmgr.go#L37-L43

It's weird that this code is duplicated, it's confusing and will make it anoying to change `LIBP2P_RCMGR`, I expect control flow (not call this function if disabled) or a field in one of the structs or an argument to be passed instead.

**Review by Jorropo — CHANGES_REQUESTED** [review]

File: cmd/ipfs/daemon.go (hunk 1 above)
Comment (gfanton) [reply cmd/ipfs/daemon.go]: it make sense, I've updated my commit, do you see a better way do this ?

**Comment by lidel** [comment]
Superseded by simpler https://github.com/ipfs/kubo/pull/9826

**Pull request closed without merging**